pkg/block/metadata/markers.go

```diff
@@ -91,6 +91,13 @@ func (n *NoCompactMark) markerFilename() string { return NoCompactMarkFilename }
 // ReadMarker reads the given mark file from <dir>/<marker filename>.json in bucket.
 func ReadMarker(ctx context.Context, logger log.Logger, bkt objstore.InstrumentedBucketReader, dir string, marker Marker) error {
 	markerFile := path.Join(dir, marker.markerFilename())
+	ok, err := bkt.Exists(ctx, markerFile)
+	if err != nil {
+		return errors.Wrapf(err, "file exists: %s", metaFile)
```

Comment:
I think you need to update the name from `metaFile` to `markerFile` to avoid the errors here.